mne/preprocessing/ieeg/_projection.py

```diff
@@ -0,0 +1,102 @@
+# Authors: Alex Rockhill <aprockhill@mailbox.org>
+#
+# License: BSD-3-Clause
+
+from os import path as op
+
+from itertools import combinations
+import numpy as np
+
+from ...io.pick import _picks_to_idx
+from ...surface import _read_mri_surface
+from ...transforms import (apply_trans, invert_transform, _cart_to_sph,
+                           _ensure_trans)
+from ...utils import verbose, get_subjects_dir, _validate_type
+
+
+@verbose
+def project_sensors_onto_brain(info, trans, subject, subjects_dir=None,
+                               picks=None, n_neighbors=10, copy=True,
+                               verbose=None):
+    """Project sensors onto the brain surface.
+
+    .. note:: This is useful in ECoG analysis for compensating for
+              "brain shift" or shrinking of the brain away from the
+              skull due to changes in pressure during the craniotomy.
+
+    .. note:: To use the brain surface, a BEM model must be created
+              e.g. using :ref:`mne watershed_bem` using the T1 or
+              :ref:`mne flash_bem` using a FLASH scan.
+
+    Parameters
+    ----------
+    %(info_not_none)s
+    %(trans_not_none)s
+    %(subject)s
+    %(subjects_dir)s
+    %(picks_base)s only ``ecog`` channels.
+    n_neighbors : int
+        The number of neighbors to use to compute the normal vectors
+        for the projection. Must be 2 or greater. More neighbors makes
+        a normal vector with greater averaging which preserves the grid
+        structure. Fewer neighbors has less averaging which better
+        preserves contours in the grid.
+    copy : bool
+        If ``True``, return a new instance of ``info``, if ``False``
+        ``info`` is modified in place.
+    %(verbose)s
+
+    Returns
+    -------
+    %(info_not_none)s
+    """
+    from scipy.spatial import distance_matrix
+    _validate_type(n_neighbors, int, 'n_neighbors')
```

Comment:
We have a specialized int-izer for this -- it does nice things like casts to an int, tolerates int-like things like `np.int32` (which the current code will not), etc.
```suggestion
    n_neighbors = _ensure_int(n_neighbors, 'n_neighbors')
```